internal/provider/resources/install/aws/aws_iam_write.go

```diff
@@ -0,0 +1,400 @@
+// Copyright (c) 2024 P0 Security, Inc
+// SPDX-License-Identifier: MPL-2.0
+
+package installaws
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/objectvalidator"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringdefault"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	"github.com/p0-security/terraform-provider-p0/internal"
+	installresources "github.com/p0-security/terraform-provider-p0/internal/provider/resources/install"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var _ resource.Resource = &AwsIamWrite{}
+var _ resource.ResourceWithImportState = &AwsIamWrite{}
+
+func NewAwsIamWrite() resource.Resource {
+	return &AwsIamWrite{}
+}
+
+type AwsIamWrite struct {
+	installer *installresources.Install
+}
+
+type AwsIamWriteLoginIdentityModel = struct {
+	Type    string  `json:"type" tfsdk:"type"`
+	TagName *string `json:"tagName" tfsdk:"tag_name"`
+}
+
+type AwsIamWriteLoginProviderMethodModel = struct {
+	Type         string `json:"type" tfsdk:"type"`
+	AccountCount *struct {
+		Type   string  `json:"type" tfsdk:"type"`
+		Parent *string `json:"parent" tfsdk:"parent"`
+	} `json:"accountCount" tfsdk:"account_count"`
+}
+
+type AwsIamWriteLoginProviderModel = struct {
+	Type             string                               `json:"type" tfsdk:"type"`
+	AppId            *string                              `json:"appId" tfsdk:"app_id"`
+	IdentityProvider *string                              `json:"identityProvider" tfsdk:"identity_provider"`
+	Method           *AwsIamWriteLoginProviderMethodModel `json:"method" tfsdk:"method"`
+}
+
+type AwsIamWriteLoginModel = struct {
+	Type     string                         `json:"type" tfsdk:"type"`
+	Identity *AwsIamWriteLoginIdentityModel `json:"identity" tfsdk:"identity"`
+	Parent   *string                        `json:"parent" tfsdk:"parent"`
+	Provider *AwsIamWriteLoginProviderModel `json:"provider" tfsdk:"provider"`
+}
+
+type AwsIamWriteModel = struct {
+	Id    string                 `tfsdk:"id"`
+	Label basetypes.StringValue  `tfsdk:"label"`
+	State basetypes.StringValue  `tfsdk:"state"`
+	Login *AwsIamWriteLoginModel `tfsdk:"login"`
+}
+
+type AwsIamWriteJson = struct {
+	Label *string                `json:"label"`
```

Comment:
```suggestion
	Label *string                `json:"label,omitempty"`
```
We should add `,omitempty` so that when we marshal to json we don't accidentally include an empty "" for label.